docs/gadget-devel/gadget-ebpf-api.md

```diff
@@ -104,29 +130,171 @@ For eBPF programs of other kinds:
 ```C
 struct sockets_value *skb_val = gadget_socket_lookup(sk, netns);
 if (skb_val != NULL) {
-        /* Access skb_val->mntns and other fields */
+	/* Access skb_val->mntns and other fields */
 }
 ```
 
 ## Enriched types
 
 When a gadget emits an event with one of the following fields, it will be
-automatically enriched.
+automatically enriched. i.e. new field(s) containing more information will be
+added to the event.
+
+### `gadget_mntns_id` and `gadget_netns_id`
+
+See [Container enrichment](#container-enrichment))
+
+### `gadget_l3endpoint_t`
+
+Represents a layer 3 endpoint. It's defined as:
 
 ```C
+struct gadget_l3endpoint_t {
+	union gadget_ip_addr_t addr_raw;
+	__u8 version; // 4 or 6
+};
+```
+
+The IP field will be translated to its string representation:
+
+```json
+  "src": {
+    "addr": "172.17.0.2",
+    "version": 4
+  },
+```
+
+### `gadget_l4endpoint_t`
+
+Represents a layer 4 endpoint. It's defined as:
+
+```C
+struct gadget_l4endpoint_t {
+	union gadget_ip_addr_t addr_raw;
+	__u16 port; // L4 port in host byte order
+	__u16 proto; // IP protocol number
+	__u8 version; // 4 or 6
+};
+```
+
+It'll produce an output like:
+
+```json
+  "src": {
+    "addr": "172.17.0.2",
+    "port": 46076,
+    "proto": 6,
```

Comment:
We should enrich that as well, hehe. (udp, tcp)

Reply:
Right, it was missing on #3076, opened #3233 to make it happen.